src/libraries/System.Memory/src/System/Buffers/SequenceReaderExtensions.Binary.cs

```diff
@@ -23,33 +23,32 @@ public static partial class SequenceReaderExtensions
         /// True if successful. <paramref name="value"/> will be default if failed (due to lack of space).
         /// </returns>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        internal static unsafe bool TryRead<T>(ref this SequenceReader<byte> reader, out T value) where T : unmanaged
+        internal static bool TryRead<T>(ref this SequenceReader<byte> reader, out T value) where T : unmanaged
         {
             ReadOnlySpan<byte> span = reader.UnreadSpan;
-            if (span.Length < sizeof(T))
+            if (span.Length < Unsafe.SizeOf<T>())
                 return TryReadMultisegment(ref reader, out value);
 
-            value = Unsafe.ReadUnaligned<T>(ref MemoryMarshal.GetReference(span));
-            reader.Advance(sizeof(T));
+            value = MemoryMarshal.Read<T>(span);
+            reader.Advance(Unsafe.SizeOf<T>());
```

Comment:
Is this method any safer?

Reply:
I can probably leave the unsafe keyword then (and sizeof)

Reply:
> The change to use Unsafe.SizeOf<T> instead of sizeof(T) doesn't seem simpler.

I normally don't replace those unless it's when `sizeof` is the only reason `unsafe` is required after my change.

Reply:
We have done a pass to delete all `Unsafe.SizeOf` uses and replaced them with `sizeof`. I do not think we want to be going back and forth.

We should wait with removing unsafe around `sizeof` until the new `unsafe` is in place in Roslyn.

Reply:
reverted